config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.java

```diff
@@ -393,7 +396,19 @@ public void oauth2LoginWithOneClientConfiguredAndRequestXHRNotAuthenticatedThenD
 		this.request.setServletPath(requestUri);
 		this.request.addHeader("X-Requested-With", "XMLHttpRequest");
 		this.springSecurityFilterChain.doFilter(this.request, this.response, this.filterChain);
-		assertThat(this.response.getRedirectedUrl()).doesNotMatch("http://localhost/oauth2/authorization/google");
+		assertThat(this.response.getStatus()).isEqualTo(401);
+	}
+
+	@Test
+	public void oauth2LoginWithOneClientConfiguredAndRequestAcceptJSONNotAuthenticatedThenDoesNotRedirectForAuthorization()
+			throws Exception {
+		loadConfig(OAuth2LoginConfig.class);
+		String requestUri = "/";
+		this.request = new MockHttpServletRequest("GET", requestUri);
+		this.request.setServletPath(requestUri);
+		this.request.addHeader("Accept", MediaType.APPLICATION_JSON);
+		this.springSecurityFilterChain.doFilter(this.request, this.response, this.filterChain);
+		assertThat(this.response.getStatus()).isEqualTo(401);
```

Comment:
This is not the expected behaviour if ONLY `oauth2Login()` is configured.

However, if `httpBasic()` is also configured then it would handle the unauthenticated request (via content negotiation) and return `401`.

Reply:
Unfortunately, I'm still not understanding why `httpBasic()` and `formLogin()` behave differently from `oauth2Login()` with regards to content negotiation. Shouldn't they all behave the same?

Reply:
> how does one setup `oauth2Login()` so it uses content negotiation and behaves like `formLogin()`

`oauth2Login()` DOES behave the same as `formLogin()`. Do you look at the tests I put together in the previous [comment](https://github.com/jgrandja/spring-security/commit/befd50a65540647115692d8b5882f5892dae3f9e#diff-1d22fe32e7c9b147d919b2d57d85b926d9251f4008b233706c0cc9cc2bdc0eacR42)?

See `formLogin_requestWhenXHRNotAuthenticatedThenRedirectToLogin` and `oauth2Login_requestWhenXHRNotAuthenticatedThenRedirectToLogin`. These tests demonstrate the same behaviour.

Reply:
@candrews 

> how does one setup `oauth2Login()` so it uses content negotiation

See this [commit](https://github.com/jgrandja/spring-security/commit/93c82779219515520086fc5420fb72dd81eb64ca) as I believe this is the configuration you are looking for